Katello/katello
# Fixes #34426 - Improve Pulp 3 sync timeout config

#### What are the changes introduced in this pull request?
Adds in configuration options for the other Pulp 3 sync timeout options and migrates the old setting.  Also corrects the use of "connect_timeout", which should've been "total_timeout" before.

#### Considerations taken when implementing this change?
 A number of users have been hitting problems were large repodata files (almost 1 GB at times) triggered the 300 second timeout.  The current recommendation to fix it is to increase the timeout to 3600.  That isn't a perfect solution, however, because then the system will wait much too long for connection errors, which have nothing to do with slow download speeds.  As such, the data download (and total_timeout) are set to 3600, but the connection-related timeouts are set to 60.  Users should not need to configure the connection timeouts, since 60 seconds is still a long time to connect.  If they do for whatever reason, however, the timeouts are there to configure.

#### What are the testing steps for this pull request?

**Test the migration**
1) Set the `sync_connect_timeout` to be something other than default
2) Patch in this PR and run the migration script
3) See that `sync_total_timeout` and `sync_sock_read_timeout` are set to the old `sync_connect_timeout`.
  -> Both of the "connect" timeouts should be the default, 60

** Check the timeouts **
1) Set up a bandwidth-limited squid proxy by installing squid and adding the following rules:
```
delay_pools 1
delay_class 1 1
delay_access 1 allow all
delay_parameters 1 1960000/1960000          # 1960 kbytes/s
```
2) Ensure the sync timeouts are at the default
3) Sync the RHEL 7 RPMs repo
4) Ensure there are no errors
5) Check the repo's Pulp remote to see what the sync timeouts are set to.
